EIPS/eip-4910.md

```diff
@@ -0,0 +1,1109 @@
+
+### Modified NFT Transfer function including required Trade data to allocate royalties
+
+The input parameters must satisfy several requirements for the NFT to be transferred AFTER the royalties have been properly distributed. Furthermore, the ability to transfer more than one token at a time is also considered.
+
+The proposal defines:
+* Input parameter validation
+* Payment Parameter Validation
+* Distributing Royalties
+* Update Royalty Account ownership with payout
+* Transferring Ownership of the NFT 
+* Removing the Payment entry in `registeredPayment` after successful transfer
+
+##### Distributing Royalties
+
+The approach to distributing royalties is to break down the hierarchical structure of interconnected Royalty Accounts into layers and then process one layer at time, where each relationship between a token and its ancestor is utilized to traverse the Royalty Account chain until the root ancestor and associated RA is reached.
+
+### Paying out Royalties to the NFT owner -- `from` address in `safeTransferFrom` function
+
+This is the final part of the proposal.
+
+There are two versions of the payout function -- a `public` function and an `internal` function.
+
+The public function has the following interface:
+```
+function royaltyPayOut (uint256 tokenId, address RAsubaccount, address payable payoutAccount, uint256 amount) public virtual nonReentrant returns (bool)
+```
+
+where we only need the `tokenId`, the RA Sub Account address, `_RAsubaccount` which is the `owner`, and the amount to be paid out, `_amount`. Note that the function has [`nonReentrant` modifier protection](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/security/ReentrancyGuard.sol), because funds are being payed out.
+
+#### Sending a Payout Payment
+
+The following steps need to be taken:
+* find the RA Sub Account based on `RAaccount` and the `subaccountPos` and extract the balance
+* extract `tokentype` from the Sub Account
+* based on the token type, send the payout payment (not exceeding the available balance)
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Data Structures: Specifications and Rationales
+
+#### Royalty Account and Royalty Sub Accounts
+
+In order to create an interconnected data structure linking NFTs to RAs that is search optimized requires to make the following additions to the global data structures of an ERC721.
+
+Note, a Royalty Account is defined as a collection of Royalty Sub Accounts linked to a meta account. This meta account is comprised of general account identifiers particular to the NFT it is linked to such as asset identifier, parent identifier etc.
+
+#### **[R1]**
```

Comment:
This header format makes the EIP rather verbose, you may consider a different format.

Reply:
@lightclient this allows each requirement to be individually referenced with links which makes discussions etc. much easier. See also the Baseline Protocol Standard as another standard using this approach to make requirements referenceable ... https://github.com/eea-oasis/baseline-standard/blob/main/core/baseline-core-v1.0-psd01.md.

Would prefer to leave it this way, please.

Reply:
resolved it using `<a name="tag_name"></a>`